### ISSUE TITLE
Entire codebase is linted for  PR, even though `VALIDATE_ALL_CODEBASE` is set to `false`

### ISSUE THREAD

**Issue by bewuethr** [github/super-linter#2548]
## Description

Setting `VALIDATE_ALL_CODEBASE` to `false` is ignored, and the entire codebase is linted.

## Expected behaviour

When I open a PR where a single file has been edited and `VALIDATE_ALL_CODEBASE` is `false`, I expect only that single file to be linted, not the entire codebase.

I have looked into a few other, similar issues (see *Additional context*) and made sure my issue is different:

- `fetch-depth` on checkout is set to `0` (though I don't think that would be required to determine what files in a PR have changed)
- the default branch is set to `main`, and my PR is into `main`

## Steps to reproduce

I see the issue in this PR: bewuethr/slack-analyzer#92; I ran locally with something like

```sh
gh repo clone bewuethr/slack-analyzer
cd slack-analyzer
git checkout update-linter

# Run with VALIDATE_ALL_CODEBASE=false
docker run \
    -e RUN_LOCAL=true \
    -e ACTIONS_RUNNER_DEBUG=true \
    -e DISABLE_ERRORS=false \
    -e ERROR_ON_MISSING_EXEC_BIT=true \
    -e LINTER_RULES_PATH=. \
    -e MULTI_STATUS=false \
    -e VALIDATE_ALL_CODEBASE=false \
    -v "$PWD":/tmp/lint \
    ghcr.io/github/super-linter:slim-v4.9.0
```

and the logs indicate that `VALIDATE_ALL_CODEBASE` was `true`, but I believe that is due to `RUN_LOCAL` being `true`.

When looking at the logs of the [workflow run](https://github.com/bewuethr/slack-analyzer/actions/runs/1889548097), I see

```txt
Run github/super-linter/slim@v4.9.0
  env:
    GITHUB_TOKEN: ***
    DEFAULT_BRANCH: main
    VALIDATE_ALL_CODEBASE: false
  
    FILTER_REGEX_EXCLUDE: img/.*

[snip]

[INFO]   --------------------------------------------
2022-02-23 20:40:12 [INFO]   Gathering user validation information...
2022-02-23 20:40:12 [INFO]   - Validating ALL files in code base...
```

and then all files being validated.

## Logs

[Logs of the job that should validate just one file, but doesn't](https://github.com/bewuethr/slack-analyzer/runs/5309902415?check_suite_focus=true)

## Additional context

- [Pull request that triggers the workflow](https://github.com/bewuethr/slack-analyzer/pull/92)
- [Relevant workflow file](https://github.com/bewuethr/slack-analyzer/actions/runs/1889548097/workflow)

### Related issues

- #1123: related to default branch; shouldn't apply, as I'm merging into the default branch
- #728: seems to be about *too few* files being linted
- #400: seems to have been about default branch
- #272: was about `VALIDATE_ALL_CODEBASE` set to `true` disabling some linters

**Comment by BretFisher** [comment]
It is weird that your [VALIDATE_ALL_CODEBASE is set to false](https://github.com/bewuethr/slack-analyzer/runs/5309902415?check_suite_focus=true#step:4:5), but later it says [Validating ALL files in code base](https://github.com/bewuethr/slack-analyzer/runs/5309902415?check_suite_focus=true#step:4:5).

I would try to keep it simple for troubleshooting and hardcode VALIDATE_ALL_CODEBASE to false and re-run.  There's also no need, from what I can see, to regex out img or manually enable linters. super-linter only gives specific file types to specific linters, so I don't see those steps saving you anything.

So, try a new commit in basic form and see if you still see it "Validating ALL files in code base":

```yaml
name: Lint code base

on:
  push:
    branches:
      - main
  pull_request:

jobs:
  lint:
    name: Lint code base
    runs-on: ubuntu-20.04
    steps:
      - name: Check out repository
        uses: actions/checkout@v2.4.0
        with:
          fetch-depth: 0

      - name: Lint code base
        uses: github/super-linter/slim@v4.9.0
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          DEFAULT_BRANCH: main
          VALIDATE_ALL_CODEBASE: false
```

BTW once you get it working, I too have it validate everything on push to main, but this is a better way to ensure "true" is returned when it's not a PR:

`VALIDATE_ALL_CODEBASE: ${{ !contains(github.event_name, 'pull_request') }}`

That prints `true` as the ENV value if it's not a pull_request event, otherwise it prints `false`. That is enough, based on your `on:` events.  No need to also check for branch, because that's already controlled by your events at top.

**Comment by bewuethr** [comment]
@BretFisher I'll try setting it to false explicitly, and also removing fetch-depth and specific languages. I was under the impression that setting specific languages to test saves time when building the file lists, but it [looks like it doesn't matter](https://github.com/github/super-linter/blob/main/lib/functions/buildFileList.sh) as long as I want to lint everything there is (and I do).

The image directory is excluded because there were false negatives with the copy-paste linter on SVGs (while other image types are [ignored by default)](https://github.com/github/super-linter/blob/e98f38ce939027f93b28348a2bea595459855436/lib/functions/buildFileList.sh#L157-L166).

**Comment by BretFisher** [comment]
Well I think you'll need fetch-depth: 0. I've always had that and it's in the example. 

Re: SVGs. Oh, yea I have a public default where I disable common "problem linters" in super-linter like copy-paste and editorconfig. I find that easier than filtering problem directories on a pre-repo basis. https://github.com/BretFisher/super-linter-workflow/blob/main/.github/workflows/super-linter.yaml#L147-L155

**Comment by bewuethr** [comment]
I don't use fetch-depth elsewhere and it works fine for me there. It's very expensive for large repos with a long history.

**Comment by bewuethr** [comment]
- Setting `VALIDATE_ALL_CODEBASE` to `false`: only lints modified files ([logs](https://github.com/bewuethr/slack-analyzer/runs/5580181816?check_suite_focus=true))
- Setting it to `${{ github.event_name == 'push' }}`: only lints modified files ([logs](https://github.com/bewuethr/slack-analyzer/runs/5580218726?check_suite_focus=true))
- Setting it back to the exact previous value, `${{ github.event_name == 'push' && github.ref_name == 'main' }}`: lints all files! ([logs](https://github.com/bewuethr/slack-analyzer/runs/5580244957?check_suite_focus=true))
- Dropping `FILTER_REGEX_EXCLUDE` and all `VALIDATE_<LANGUAGE>` settings: lints all files ([logs](https://github.com/bewuethr/slack-analyzer/runs/5580273744?check_suite_focus=true))

At this point, the pull request synchronize events seems to have stopped firing...

![image](https://user-images.githubusercontent.com/8521043/158732675-d0afa33d-f4e7-4248-be90-6a31b39ab948.png)

I've added a manual trigger, which *should* behave the same as a PR event if I read [this](https://github.com/github/super-linter/blob/e277edca06d0850200869e6552a94da09f95cb3a/lib/functions/buildFileList.sh#L126) correctly.

- Dropping `fetch-depth`: lints all files ([logs](https://github.com/bewuethr/slack-analyzer/runs/5580388876?check_suite_focus=true))

  Regarding this being required or not: it seems to me that it's not required, despite the README; the necessary steps happen here:

  https://github.com/github/super-linter/blob/e277edca06d0850200869e6552a94da09f95cb3a/lib/functions/buildFileList.sh#L73-L89

On the other hand, in bewuethr/slack-analyzer#98, it works as expected ([logs](https://github.com/bewuethr/slack-analyzer/runs/5580550943?check_suite_focus=true)) with the settings that I have adopted recently.

In other words, there's something wrong, but it doesn't bother me any longer, so unless somebody is deeply interested in solving this, I consider it closed.

**Comment by bewuethr** [comment]
Okay, last comment before I truly let this be, but the nondeterministic behaviour is driving me crazy. I have two workflow runs on PRs:

- PR 1, bewuethr/slack-analyzer#98, modifies two files (including the workflow file itself), and uses [this workflow definition](https://github.com/bewuethr/slack-analyzer/actions/runs/1996543584/workflow) ([logs](https://github.com/bewuethr/slack-analyzer/runs/5580550943?check_suite_focus=true))
- PR 2, bewuethr/slack-analyzer#99, modifies one file (just the workflow file itself), and uses [this workflow definition](https://github.com/bewuethr/slack-analyzer/actions/runs/1996579382/workflow) ([logs](https://github.com/bewuethr/slack-analyzer/runs/5580639109?check_suite_focus=true))

This is the diff between the to workflow definitions:

```diff
28c28
<           VALIDATE_ALL_CODEBASE: ${{ github.event == 'push' }}
---
>           VALIDATE_ALL_CODEBASE: ${{ github.event_name == 'push' }}
```

Since both workflow runs are triggered by pull requests, `VALIDATE_ALL_CODEBASE` is `false` for both, and both say

```txt
 - Only validating [new], or [edited] files in code base...
```

in their logs.

But why on earth does the first workflow successfully find the two files and lint them, and the second workflow says this?

```txt
2022-03-17 04:12:19 [WARN]   No files were found in the GITHUB_WORKSPACE:[/github/workspace] to lint!
```
